iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -492,8 +492,7 @@ bool PortManager::isCompatiblePubSub(const PublisherPortRouDiType& publisher,
         !(pubOpts.subscriberTooSlowPolicy == popo::ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA
           && subOpts.queueFullPolicy == popo::QueueFullPolicy::BLOCK_PRODUCER);
 
-    const bool historyRequestIsCompatible =
-        !subOpts.requiresPublisherHistorySupport || subOpts.historyRequest <= pubOpts.historyCapacity;
+    const bool historyRequestIsCompatible = !subOpts.requiresPublisherHistorySupport || pubOpts.historyCapacity > 0;
```

Comment:
This is the actual change that requires the documentation, examples and tests to be adapted in all the other files.

Reply:
For me this new behavior makes even more sense. Since we cannot guarantee to deliver any history (the publisher might just be started), this more relaxed requirement mirror the reality a bit better.

Reply:
Yes but there is a difference between may not have enough data and will never have enough data.